### PR TITLE
Guest OS Support: add JeOS/Fedora 22/23 [v3]

### PR DESCRIPTION
**Pull request by clebergnu**
This PR adds (late) support for Fedora 22 and 23, and also the JeOS guests based on it.

New JeOS images (based on Fedora 23), based on this configuration/kickstart have been pushed to the official asset repo.

--
Changes from v2 (#439):
   * Only overwrite asset files if newer than the user's version

Changes from v1:
   * Added development related packages to F22 and F23
   * Added Python 2 packages on F23
   * Set the JeOS.23 as the new default guest os
   * Removed the JeOS.20 asset (download) definition file
   * Fix the `data_dir.get_download_dir()` implementation and usage

### CODE REVIEW

File: virttest/data_dir.py [hunk 1]
```diff
@@ -214,8 +216,22 @@ def get_tmp_dir(public=True):
...
+
+    In sync means that newer or non-existing asset definition files
+    are copied to the user's directory.  If the user's directory
+    contains newer files, they're not overwritten.  This sync behavior
+    is based on :func:`distutils.file_util.copy_file` parameter
+    "update".
+
+    :returns: a writable directory populated with asset definition files
+    """
+    dir_util.copy_tree(BASE_DOWNLOAD_DIR, DOWNLOAD_DIR, update=1)
```
Comment (ldoktor) [inline virttest/data_dir.py]:
```
    If 'update' is true, 'src' will only be copied if 'dst' does not exist,
    or if 'dst' does exist but is older than 'src'.
```

That means if the user updates the file and then we update the file in git, it'll be overwritten. How about the `vt-bootstrap --vt-update-config` approach, wouldn't that be better?

Actually I forgot how `--vt-update-config` works, so instead I think it should be yet another (interactive) step in `vt-bootstrap` and if the `--vt-update-config` is used it should force-update. At least that's what I'd expect it to do as these are (asset) conifg files.

Comment (ldoktor) [inline virttest/data_dir.py]: Btw this is the only issue in this PR, how about merging it first and deal with this later?

**Comment by clebergnu** [comment]
Closing in favor of #472

File: shared/unattended/Fedora-22.ks [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+ECHO "OS install is completed"
+grubby --remove-args="rhgb quiet" --update-kernel=$(grubby --default-kernel)
+dhclient
+chkconfig sshd on
+iptables -F
+systemctl mask tmp.mount
+echo 0 > /selinux/enforce
+sed -i "/^HWADDR/d" /etc/sysconfig/network-scripts/ifcfg-eth0
+dnf -y groupinstall c-development development-tools
+ECHO 'Post set up finished'
```
Comment (ldoktor) [inline shared/unattended/Fedora-22.ks]: no changes here, ack

File: shared/unattended/JeOS-22.ks [hunk 3]
```diff
@@ -0,0 +1,189 @@
...
+sed -i "/^HWADDR/d" /etc/sysconfig/network-scripts/ifcfg-eth0
+sed -i -e "s,^GRUB_TIMEOUT=.*,GRUB_TIMEOUT=0," /etc/default/grub
+grub2-mkconfig > /etc/grub2.cfg
+yum install -y hdparm ntpdate qemu-guest-agent
+yum clean all
+mkdir -p /var/log/journal
+sed -i -e 's/\#SystemMaxUse=/SystemMaxUse=50M/g' /etc/systemd/journald.conf
+dd if=/dev/zero of=/fill-up-file bs=1M
+rm -f /fill-up-file
+ECHO 'Post set up finished'
```
Comment (ldoktor) [inline shared/unattended/JeOS-22.ks]: no changes, ack

File: shared/unattended/Fedora-23.ks [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+ECHO "OS install is completed"
+grubby --remove-args="rhgb quiet" --update-kernel=$(grubby --default-kernel)
+dhclient
+chkconfig sshd on
+iptables -F
+systemctl mask tmp.mount
+echo 0 > /selinux/enforce
+sed -i "/^HWADDR/d" /etc/sysconfig/network-scripts/ifcfg-eth0
+dnf -y groupinstall c-development development-tools
+ECHO 'Post set up finished'
```
Comment (ldoktor) [inline shared/unattended/Fedora-23.ks]: no changes, ack

File: shared/unattended/JeOS-23.ks [hunk 5]
```diff
@@ -0,0 +1,188 @@
...
+sed -i "/^HWADDR/d" /etc/sysconfig/network-scripts/ifcfg-eth0
+sed -i -e "s,^GRUB_TIMEOUT=.*,GRUB_TIMEOUT=0," /etc/default/grub
+grub2-mkconfig > /etc/grub2.cfg
+yum install -y hdparm ntpdate qemu-guest-agent
+yum clean all
+mkdir -p /var/log/journal
+sed -i -e 's/\#SystemMaxUse=/SystemMaxUse=50M/g' /etc/systemd/journald.conf
+dd if=/dev/zero of=/fill-up-file bs=1M
+rm -f /fill-up-file
+ECHO 'Post set up finished'
```
Comment (ldoktor) [inline shared/unattended/JeOS-23.ks]: no changes, ack

File: virttest/defaults.py [hunk 6]
```diff
@@ -18,7 +18,7 @@ def get_default_guest_os_info():
     Gets the default asset and variant information
     TODO: Check for the ARCH and choose corresponding default asset
     """
-    return {'asset': 'jeos-21-64', 'variant': 'JeOS.21'}
+    return {'asset': 'jeos-23-64', 'variant': 'JeOS.23'}
 
 
 DEFAULT_GUEST_OS = get_default_guest_os_info()['variant']
```
Comment (ldoktor) [inline virttest/defaults.py]: I agree, out of curiosity, what JeOS images are there? (I dislike it does not support browsing) I noticed 21 and 23.

File: shared/unattended/JeOS-23.ks (hunk 5 above)
Comment (ldoktor) [inline shared/unattended/JeOS-23.ks]: OMG I'm revieweing the old version... OK me restart :smile:

**Pull request closed without merging**